configure.ac

```diff
@@ -227,6 +227,17 @@ AM_COND_IF([ENABLE_MANPAGES], [
 	AS_IF([test "x$RST2MAN" = "x"], [AC_MSG_ERROR([[--enable-manpages requires rst2man]])])
 ])
 
+AC_ARG_ENABLE([bash-completion],
+	AS_HELP_STRING([--enable-bash-completion@<:@=DIR@:>@], [install bash completion script @<:@default=no@:>@]),
+	[bashcompdir="${enableval}"],
+	[bashcompdir=$($PKG_CONFIG --variable=completionsdir bash-completion)]
```

Comment:
The action-if-not-given is used when there is no `--enable-xxx` given, not when there is no argument given to this option. Also, do not add `default=no` to help message, someone (e.g. me) might read it as DIR=no by default, and since the option will be `--enable` (or `--with`) it implies that the it is disabled by default.

In the previous revision there was also a check `PKG_CHECK_MODULES([BASH_COMPLETION], [bash-completion >= 2.0]`. I would keep it here, because `pkg-config --variable=completionsdir bash-completion`, in case where there is no bash-completion package, will return error.

Reply:
> The action-if-not-given is used when there is no --enable-xxx given

Oops - sorry. I thought I'd tested without the option - obviously not :disappointed:  

> do not add default=no to help message

OK

> there was also a check PKG_CHECK_MODULES([BASH_COMPLETION]

The thing is that bash-completion is not needed on the build machine at all, so it seems unnecessary to fail the configure step for this. Perhaps we should drop the pkg-config parts entirely? There is no way for the build to know what the runtime host will be, nor how it is configured (except in the special case that the build and host are the same). For example, I build on my laptop, but run bluealsa on various Raspberry Pis around the house. We could just default to bash-completion's built-in default and let the user care about the runtime hosts distribution's and admin's "quirks"?

Reply:
> The thing is that bash-completion is not needed on the build machine at all, so it seems unnecessary to fail the configure step for this.

Yes, failing is unnecessary, but you will not have to fail. If bash-completion package is available use pkg-config for retrieving path, otherwise use built-in one. And of course use pkg-config or built-in only in case where user has not specified the path (`$enableval = yes`).